### PR TITLE
add new analysis routes & improve scrolling

### PR DESCRIPTION
**Pull request by maracuja-juice**
New features:
- new route `/analysis/latest` -> this will display the last played game that was locally saved
  This will make it possible to add a button when the game is finished that redirects to the analysis view.
- new route `/analysis/:date` -> this will display the game that was saved at that time
  This will make it possible to go from the `/analysis` overview to the analysis view also for games for which saving was not successful or that weren't saved yet. The date has to be an ISO formatted string.

Fixed Bugs:
- On bigger screens the automatic scrolling had an offset that made it impossible to view the top item when automatically scrolling up again. Now the top item is not the selected item anymore which makes the experience on all screens better.

Other:
- Prettier now also formats CSS files.

related to #103

### CODE REVIEW

File: src/state/analysisGame/analysisGameSaga.ts [hunk 1]
```diff
@@ -1,22 +1,55 @@
...
   // TODO if no results in selector: try over network
 
-  if (finishedGame !== undefined) {
-    yield put(setAnalysisGame(finishedGame));
+  if (analysisGame !== undefined) {
+    yield put(setAnalysisGame(analysisGame));
+  }
+}
+
+function* loadFinishedGameByDate(action: GenericAction) {
```
Comment (maracuja-juice) [inline src/state/analysisGame/analysisGameSaga.ts]: These 3 functions do very similar things right now. Should / Can we make this simpler?

Comment (RohrerF) [reply src/state/analysisGame/analysisGameSaga.ts]: maybe make the if(){...} in a seperate function. But i think this just makes everything more complicated. I would leave it as is

File: src/components/App.test.tsx [hunk 2]
```diff
@@ -1,11 +0,0 @@
...
-import { shallow } from 'enzyme';
-import { App } from './App';
-
-describe('App', () => {
-  it('should match snapshot', () => {
-    const app = shallow(<App />);
-
-    expect(app).toMatchSnapshot();
-  });
-});
```
Comment (RohrerF) [inline src/components/App.test.tsx]: Why removed?

File: src/components/App.tsx [hunk 3]
```diff
@@ -13,8 +14,17 @@ export class App extends React.Component {
     return (
       <Router>
         <Switch>
-          <Route path="/analysis/:id" exact={true} component={AnalysisGame} />
-          <Route path="/" component={Game} />
+          <Route
+            path={appRoutes.AnalysisLatest.path}
```
Comment (RohrerF) [inline src/components/App.tsx]: Use string properties to avoid always using `.path`

File: src/routes/routes.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const appRoutes = {
+  AnalysisLatest: {
+    path: '/analysis/latest',
```
Comment (RohrerF) [inline src/routes/routes.ts]: Path property unnecessary if no other properties are planed

File: src/state/analysisGame/analysisGameSaga.ts (hunk 1 above)
Comment (maracuja-juice) [reply src/state/analysisGame/analysisGameSaga.ts]: I don't think that would add much value 🤔

File: src/components/App.test.tsx (hunk 2 above)
Comment (maracuja-juice) [reply src/components/App.test.tsx]: I don't think that this snapshot test added much value.
I'll add a "not exploding test" instead.

File: src/routes/routes.ts (hunk 4 above)
Comment (maracuja-juice) [reply src/routes/routes.ts]: You're right.
First wanted to add some other properties aswell but I think that your proposal works better. I don't have any other properties planned.

**Pull request merged**